### PR TITLE
[INFRA] libc++ CI

### PR DESCRIPTION
**Pull request by eseiler**

### CODE REVIEW

File: .github/workflows/ci_macos.yml [hunk 1]
```diff
@@ -66,8 +74,7 @@ jobs:
           mkdir build
           cd build
           cmake .. -DCMAKE_BUILD_TYPE=${{ matrix.build_type }} \
-                   -DCMAKE_CXX_FLAGS="-Werror -Wno-interference-size" \
-                   -DCHOPPER_NATIVE_BUILD=OFF
```
Comment (smehringer) [inline .github/workflows/ci_macos.yml:70]: why is `-DCHOPPER_NATIVE_BUILD=OFF` removed completely?

File: test/api/layout/execute_layout_test.cpp [hunk 2]
```diff
@@ -321,6 +321,104 @@ TEST(execute_test, many_ubs)
                                       "#LOWER_LEVEL_IBF_26 fullest_technical_bin_idx:0\n"
                                       "#LOWER_LEVEL_IBF_27 fullest_technical_bin_idx:0\n"
                                       "#USER_BIN_IDX\tTECHNICAL_BIN_INDICES\tNUMBER_OF_TECHNICAL_BINS\n"
+#ifdef _LIBCPP_VERSION // seqan::hibf::sketch::toolbox::sort_by_cardinalities is not stable
```
Comment (smehringer) [inline test/api/layout/execute_layout_test.cpp:324]: but it is deterministic within the same compiler?

what a code bloat :sweat:

**Review by smehringer — APPROVED** [review]

File: test/api/layout/execute_layout_test.cpp (hunk 2 above)
Comment (eseiler) [reply test/api/layout/execute_layout_test.cpp:324]: It's deterministic/constitent within each stl (libc++/libstdc++).

The alternative would be to use:
* `std::ranges::stable_sort` in the HIBF method, but this has a potentially higher complexity, and also needs additional memory
* make each size distinct. E. G. 200,201,202,..

Comment (smehringer) [reply test/api/layout/execute_layout_test.cpp:324]: nah. it's fine.

**Pull request merged**